pkg/engine/lifecycle_test.go

```diff
@@ -139,7 +139,7 @@ func (j *Journal) Snap(base *deploy.Snapshot) *deploy.Snapshot {
 
 	manifest := deploy.Manifest{}
 	manifest.Magic = manifest.NewMagic()
-	return deploy.NewSnapshot(manifest, resources)
+	return deploy.NewSnapshot(manifest, resources, nil)
```

Comment:
Nit: it would be good to construct a real set of pending operations by keeping track of what has and hasn't been retired in the loop above.